civicrm.php

```diff
@@ -292,6 +298,42 @@ public function __wakeup() {
     _doing_it_wrong(__FUNCTION__, __('Please do not serialize CiviCRM_For_WordPress', 'civicrm'), '4.4');
   }
 
+  protected function is_iframe(): bool {
+    if (is_admin()) {
+      // We intend to process IFRAMEs through WP-frontend not WP-backend.
+      // This is separate from the actual content of the page (which isn't really WP-frontend or WP-backend).
+      return FALSE;
+    }
+
+    // return (1 == get_query_var('_cvwpif')); // Too Early
+    return !empty($_REQUEST['_cvwpif']);
+    // return 'iframe' === $_SERVER['HTTP_SEC_FETCH_DEST']; // Harder to test. And Safari only gained support a year ago.
+  }
+
+  protected function activate_iframe(): void {
+    // By default, internal links should stay in the iframe.
+    $GLOBALS['civicrm_url_defaults'][]['scheme'] = 'iframe';
+
+    // Strict browsers (eg Safari) will quietly disregard cookies when loading a page within an IFRAME.
+    // But it's quite awkward to test behavior with two variables (browser-type and page-context).
+    // For consistent testing/UX, we force similar behavior for any request with IFRAME-style URL.
+
+    // Variant A: Disregard specific pieces
```

Comment:
@totten Looking at this section I'd like to clarify:

- if vwe determine we are in an iframe https://github.com/civicrm/civicrm-wordpress/pull/336/files#diff-302f8b24886ed21286616e661003877a59e8f3ca6d6af0437574f0bdee5b0a10R261 which depends on https://github.com/civicrm/civicrm-wordpress/pull/336/files#diff-302f8b24886ed21286616e661003877a59e8f3ca6d6af0437574f0bdee5b0a10R309
- we then remove the three following 'cookies'

If I have that correct, how do we then reset them after form submission?  
-

Reply:
@kcristiano Good question. I think you point to the right bits of code (basically: the presence of `$_REQUEST['_cvwpif']` determines if the current request is part of an IFRAME).

In Variant A, it doesn't actually delete the cookies. From an HTTP POV, the cookies are still transmitted and valid. But within the IFRAME request, we skip the WordPress function which reads the cookies for authentication. So as far as WordPress is concerned, within that IFRAME request, the cookies are inert -- leaving the user as anonymous.

Then on any other request, the cookies can still behave normally.

(IIRC, for Civi-WP environments, variants B+C also preserve the cookie from HTTP POV.)

Reply:
Thanks Tim.   As soon as I complete testing we can make the core PR merge ready.   

I've also asked @christianwach for their input.   

It'd be nice to get this merged just after you branch for the next RC

Reply:
Confirming tested and working on remote WordPress site (different server) from WordPress/CiviCRM site.